src/views/ItemsView.vue

```diff
@@ -1,6 +1,286 @@
+<script setup>
+import { useItemsStore } from '@/stores/items.js'
+import { computed, ref, onMounted, onBeforeUnmount } from 'vue'
+
+// Items Store
+const itemsStore = useItemsStore()
+const items = computed(() => itemsStore.items)
+
+const selectedCategories = ref([])
+const isDropdownOpen = ref(false)
+
+// Extracting unique categories from items
+const uniqueCategories = computed(() => {
+  const categories = items.value.flatMap((item) => item.category)
+  return [...new Set(categories)]
+})
+
+// Filter items based on selected categories
+const filteredItems = computed(() => {
+  if (selectedCategories.value.length === 0) return items.value
+  return items.value.filter((item) =>
+    item.category.some((cat) => selectedCategories.value.includes(cat)),
+  )
+})
+
+// Close dropdown when click outside
+const handleClickOutside = (event) => {
+  const dropdownMenu = document.querySelector('.dropdown-menu')
+  const dropdownButton = document.querySelector('.dropdown-button')
+  if (
+    dropdownMenu &&
+    !dropdownMenu.contains(event.target) &&
+    !dropdownButton.contains(event.target)
+  ) {
+    isDropdownOpen.value = false
+  }
+}
+
+onMounted(() => {
+  document.addEventListener('mousedown', handleClickOutside)
+})
+
+onBeforeUnmount(() => {
+  document.removeEventListener('mousedown', handleClickOutside)
+})
```

Comment:
You can remove this block and add `@click="isDropdownOpen = false"` to line 49 and `@click.stop` to line 54 and 57 to avoid propagation